source/MaterialXCore/Look.cpp

```diff
@@ -233,24 +216,48 @@ void LookGroup::append(const LookGroupPtr& sourceGroup, const string& appendAfte
             }
         }
     }
+}
 
-    // Append the source list
+static void appendLooks(StringVec& destLookList, const StringVec& sourceLookList, const StringSet& destLookSet)
+{
     for (const string& lookName : sourceLookList)
     {
         if (!destLookSet.count(lookName))
         {
             destLookList.push_back(lookName);
         }
     }
+}
 
-    // Append anything remaining from destination list
+static void appendPostLooks(StringVec& destLookList, const StringVec& postList)
+{
     if (postList.size())
     {
         for (const string& lookName : postList)
         {
             destLookList.push_back(lookName);
         }
     }
+}
+
+void LookGroup::appendLookGroup(const LookGroupPtr& sourceGroup, const string& appendAfterLook)
+{
+    string sourceLooks = sourceGroup->getLooks();
+    if (sourceLooks.empty())
+    {
+        return;
+    }
+
+    // If look already exists in the look list then append if
+    // not skipping duplicates
+    const StringVec& sourceLookList = splitString(sourceLooks, ARRAY_VALID_SEPARATORS);
+    StringVec destLookList = splitString(getLooks(), ARRAY_VALID_SEPARATORS);
+    const StringSet destLookSet(destLookList.begin(), destLookList.end());
+
+    StringVec postList;
+    appendPreLooks(postList, destLookList, destLookSet, appendAfterLook);
```

Comment:
Broke this up into 3 helper methods (could have made it just one I suppose) that are used by both appendLookGroup and appendLook.